### ISSUE TITLE
Floor not simplifying/evaluating final answer

### ISSUE THREAD

**Issue by 0sidharth** [sympy/sympy#20865]
```py
In [1]: from sympy import __version__

In [2]: __version__
Out[2]: '1.8.dev'

In [3]: t = sqrt(3) + 1

In [4]: r = (1 / t).simplify()

In [5]: s = t - 1 / r

In [6]: s
Out[6]: 
     1             
- ──────── + 1 + √3
    1   √3         
  - ─ + ──         
    2   2          

In [7]: s.simplify()
Out[7]: 0

In [8]: floor(s)
Out[8]: 
     1             
- ──────── + 1 + √3
    1   √3         
  - ─ + ──         
    2   2          

In [9]: ceiling(s)
Out[9]: 
     1             
- ──────── + 1 + √3
    1   √3         
  - ─ + ──         
    2   2          

In [10]: frac(s)
Out[10]: 0
```
It would be nice to have `floor(s)` and `ceiling(s)` output 0.

**Comment by 0sidharth** [comment]
Returning `arg.simplify()` here

https://github.com/sympy/sympy/blob/bf2e1a4e166485d78ea182547824978064c61ec2/sympy/functions/elementary/integers.py#L28-L29

could be a fix perhaps? Or maybe simplifying `arg` before calling `_eval_number` (not sure if this will mess up other cases) - 

https://github.com/sympy/sympy/blob/bf2e1a4e166485d78ea182547824978064c61ec2/sympy/functions/elementary/integers.py#L22-L24

**Comment by oscargus** [comment]
Is this causing a problem inside of SymPy (this is the reason that some other expression blows up)? If not, as you showed, calling `simplify` on the result will give you 0.

In general, there is a trade-off how much time one should spend simplifying inside of functions, slowing the particular function down but possibly speeding other computations up, and keeping that particular function fast at the possible risk of other becoming slower because of expression blowing up.

So if it is part of another computation it may be better to call simplify (or something simpler), there, rather than slowing all computations of floor and ceiling down. But it may also be a good idea to introduce some simpler "simplification" to handle most cases inside floor and ceiling.

**Comment by 0sidharth** [comment]
No, I don't think this was encountered during debugging. It's just a standalone issue. I'd just thought that having these equations simplified would be nice, although if there is a slowdown across all calls to floor and ceiling due to this, perhaps its not needed.

There is indeed a trade-off, so this issue may be closed if its deemed that this utility isn't worth the slowdown.

**Comment by oscargus** [comment]
Thanks for the clarification. The thing here is that it seems like `is_integer` evaluates to True since it is zero. It would be nice to be able to get that result somehow as it is already determined...
